### DISCUSSION TITLE
Feature Request: Enhanced Edge Styling and Interactive Node-to-Node Edge Creation

### DISCUSSION THREAD

**Discussion by ShounakJoshi07** [reaviz/reagraph#228] [Ideas]
Hello Reagraph Maintainers and Community,

I've been working with Reagraph and find it an excellent tool for graph visualization in React. I would like to suggest a few enhancements that could benefit users who need more customization and interactive capabilities:

1. Colored Edges: It would be beneficial to support custom colors for edges directly through the API. This can help in distinguishing between different types of relationships in the graph more clearly.
2. Custom Edge Styling: Adding support for different edge styles, such as dashed, dotted, or any other CSS-based line patterns, would increase the flexibility of the library in representing various types of connections.
3. Interactive Edge Creation: Implementing a feature where users can drag from one node to another to create an edge interactively would enhance the user experience, making it more intuitive to build and modify graphs dynamically.

These features could potentially make Reagraph more versatile and suitable for a wider range of graph visualization needs. I believe these enhancements could also attract new users who require these specific capabilities.

Thank you for considering these suggestions. I am looking forward to any discussion or feedback regarding the feasibility of these ideas.

**Comment by crimson-med** [comment]
This would be amazing to have

**Comment by AntonJames-Sistence** [comment]
Hi @ShounakJoshi07 ,

Thanks a lot for the thoughtful suggestions and for sharing your experience with Reagraph! We really appreciate the feedback. Here’s a quick update on each point:

1. **Colored Edges** - Should be available under fill prop on the Edge.

2. **Custom Edge Styling** – This should already be supported in the latest version.

3. **Interactive Edge Creation** – This one is a bit more complex to implement, so it may take more time.

Thanks again for taking the time to suggest these improvements. They’re very much aligned with our goal of making Reagraph more flexible and user-friendly!

**Reply by tri-scott** [reply]
Any progress on interactive edge creation? Something as simple as holding down shift and dragging from one node to another would be sufficient, maybe even ideal for many people. The only UI element would be a temporary arrow. I think Cytoscape.js implements this with a _phantom node_. Basically they create an invisible node with an arrow from the source node to the invisible node. Dragging around just drags the invisible node until it coincides with a target node.
